### PR TITLE
Firewall edits

### PR DESCRIPTION
**Pull request by kaihoffman**
Implements Firewall/Firewallrule methods:

- Create Firewall
- List Firewalls
- Remove Firewall

- New Rule
- List Rules
- Delete Rule

### CODE REVIEW

File: lib/firewall.rb [hunk 1]
```diff
@@ -1,5 +1,85 @@
 module CivoCLI
   class Firewall < Thor
+    desc "create firewall_name", "Create a new firewall"
+    def create(firewall_name)
+      CivoCLI::Config.set_api_auth
 
+      Civo::Firewall.create(name: firewall_name)
+      puts "        Created firewall #{firewall_name.colorize(:green)}"
```
Comment (andyjeffries) [inline lib/firewall.rb]: Same problem as before - crazy indentation before the output...

File: lib/firewall.rb [hunk 2]
```diff
@@ -1,5 +1,85 @@
...
+    rescue Flexirest::HTTPException => e
+      puts e.result.reason.colorize(:red)
+      exit 1
+    end
+
+    desc "remove firewall_ID", "Removes a firewall with Firewall ID provided"
+    def remove(firewall_ID)
+      CivoCLI::Config.set_api_auth
+      Civo::Firewall.remove(id: firewall_ID)
+      puts "        Removed firewall #{firewall_ID.colorize(:red)}"
```
Comment (andyjeffries) [inline lib/firewall.rb]: Again

File: lib/firewall.rb [hunk 3]
```diff
@@ -1,5 +1,85 @@
...
+    rescue Flexirest::HTTPException => e
+      puts e.result.reason.colorize(:red)
+      exit 1
+    end
+
+    desc "new_rule", "Create new rule on firewall firewall_name called rule_name with required arguments"
```
Comment (andyjeffries) [inline lib/firewall.rb]: Not a fan of using _ in command names, `civo firewall new-rule` looks better.

File: lib/firewall.rb [hunk 4]
```diff
@@ -1,5 +1,85 @@
...
+    def new_rule
+      CivoCLI::Config.set_api_auth
+
+      if !options[:end_port]
+        Civo::FirewallRule.create(firewall_id: options[:firewall_id], start_port: options[:start_port], end_port: options[:start_port], cidr: options[:cidr], direction: options[:direction], label: options[:label])
+      end
+      if options[:end_port]
+        Civo::FirewallRule.create(firewall_id: options[:firewall_id], start_port: options[:start_port], end_port: options[:end_port], cidr: options[:cidr], direction: options[:direction], label: options[:label])
+      end
+      puts "        New rule" + (options[:label].nil? ? "" : " #{options[:label].colorize(:green)}") + " created"
```
Comment (andyjeffries) [inline lib/firewall.rb]: Again crazy spacing.

File: lib/firewall.rb (hunk 3 above)
Comment (andyjeffries) [reply lib/firewall.rb]: Not sure how that works with method names though, you may need to research that...

File: lib/firewall.rb [hunk 5]
```diff
@@ -1,5 +1,85 @@
...
+      puts e.result.reason.colorize(:red)
+      exit 1
+    end
+
+    desc "delete_rule firewall_id rule_id", "Deletes rule with rule_id from firewall with firewall_id"
+    def delete_rule(firewall_id, rule_id)
+      CivoCLI::Config.set_api_auth
+
+      Civo::FirewallRule.remove(firewall_id: firewall_id, id: rule_id)
+      puts "        Removed Firewall rule #{rule_id.colorize(:red)}"
```
Comment (andyjeffries) [inline lib/firewall.rb]: Spacing...

**Review by andyjeffries — CHANGES_REQUESTED** [review]

File: lib/firewall.rb (hunk 1 above)
Comment (kaihoffman) [reply lib/firewall.rb]: Sorry to miss this from the fixes. Changing all.

Comment (andyjeffries) [reply lib/firewall.rb]: 👍

File: lib/firewall.rb (hunk 3 above)
Comment (kaihoffman) [reply lib/firewall.rb]: Ruby doesn't allow hyphens in method names. Option would be to have it be called `newrule` for: `civo firewall newrule --firewall_id=98babh3...`

Comment (andyjeffries) [reply lib/firewall.rb]: LOL! I know that. I meant maybe thor will allow you to use a hyphenated command internally call the underscore version ;-)

Comment (andyjeffries) [reply lib/firewall.rb]: https://stackoverflow.com/questions/40064574/hyphenated-subcommand-in-thor

**Pull request merged**